### PR TITLE
#114 Add missing <op> with the name of the method being called.

### PR DESCRIPTION
**Pull request by rok-povsic**

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#159](https://github.com/yegor256/jpeek/issues/159) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#159](https://github.com/yegor256/jpeek/issues/159) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by amihaiemil** [comment]
@rok-povsic There are build failures, please fix them. I'll have a look afterwards :)

**Comment by rok-povsic** [comment]
@amihaiemil Build now passes.

**Comment by amihaiemil** [comment]
@rok-povsic can you check the conflict?

**Comment by amihaiemil** [comment]
@skapral ping ;)

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by rok-povsic** [comment]
@amihaiemil I've resolved the conflicts with the master branch. My task has been basically reimplemented in the master branch meanwhile the PR has been open. Anyway, the task is done and next puzzles open.

**Comment by amihaiemil** [comment]
@rok-povsic If the task is already done and the PR doesn't solve anything, then I guess you can close it :-?

**Comment by rok-povsic** [comment]
@amihaiemil I wouldn't agree with that. It's really not my fault that someone has decided to reimplement the functionality I've written already. He could've just copied the code from my commit (it has been pushed before any master branch changes).

Also, I've added a puzzle and tests that are not in master branch, so it's a step further in any case.

**Comment by amihaiemil** [comment]
@rok-povsic I don't even understand, how and where has the ticket already been solved. I see you were the only assignee so far, so how has it been solved somewhere else?

**Comment by rok-povsic** [comment]
@amihaiemil The same approach I've done has also been done [here](https://github.com/yegor256/jpeek/pull/153/files) and merged yesterday. See the `visitMethodInsn` implementation. I've written basically the exact same code. After resolving the conflict with the master branch, it seems like I've done little work, just because the other PR has been merged into master before this PR.

If this PR is rejected, there would be a precedent that even though you do the work, you can still get no reward because someone merged in the same changes before you (or could even take the code you've commited, add it to his own PR, and if gets merged before you, steal your work). 

But this PR in any case has added tests + done the next puzzle.

File: src/main/java/org/jpeek/Skeleton.java [hunk 1]
```diff
@@ -267,6 +268,18 @@ public FieldVisitor visitField(final int access,
...
         }
+        // @todo #114:30min Find and include the name of the
+        //  variable the method is called by, if possible.
+        //  Currently, only name of the method is retrieved.
+        //  Example 1: `Bar.NAME.length();`
+        //  - Here, `length` is retrieved from the param `name`, but the word
+        //  - NAME` is not included in any of the `visitMethodInsn`
+        //  - arguments.
+        //  Example 2: `src.length();`
+        //  - Here, `length` is retrieved from param `name`,
```
Comment (amihaiemil) [inline src/main/java/org/jpeek/Skeleton.java]: @rok-povsic I think here you wanted to say ``'length' is retrieved from param 'src'``

File: src/main/java/org/jpeek/Skeleton.java [hunk 2]
```diff
@@ -267,6 +268,18 @@ public FieldVisitor visitField(final int access,
...
+        //  - but the word `src` is not included in any of
+        //  - the `visitMethodInsn` arguments.
+        //  This is currently implemented in `visitMethodInsn` down below.
```
Comment (amihaiemil) [inline src/main/java/org/jpeek/Skeleton.java]: @rok-povsic What do you mean, "this is currently implemented in the method below"? If it is implemented, what is the puzzle for? I think you should rephrase this sentence :)

File: src/main/java/org/jpeek/Skeleton.java [hunk 3]
```diff
@@ -80,7 +80,8 @@
         "PMD.CyclomaticComplexity",
         "PMD.StdCyclomaticComplexity",
         "PMD.ModifiedCyclomaticComplexity",
-        "PMD.TooManyMethods"
+        "PMD.TooManyMethods",
+        "PMD.ExcessiveMethodLength"
```
Comment (amihaiemil) [inline src/main/java/org/jpeek/Skeleton.java]: @rok-povsic Is this still required?

**Comment by amihaiemil** [comment]
@rok-povsic also, please remove the keyword "fix" from the description. We shouldn't close tickets automatically, they should be closed by their authors always.

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by rok-povsic** [comment]
@amihaiemil Thanks for looking into this, and for the comments which I've fixed (as you've seen). Yes, I also agree this shouldn't happen often.

**Comment by rok-povsic** [comment]
@yegor256 Ping.

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+540](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+9634](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#159](https://github.com/yegor256/jpeek/issues/159) is now out of scope

**Comment by llorllale** [comment]
I think I should've been clearer in my description for #114. At the time I submitted the bug I was more interested in access to `Bar.NAME` than to the method call `Bar.NAME.length()`.

**Pull request merged**